File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution(object):
    def longestOnes(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        left = 0
        res = 0

        for right in range(len(nums)):
            if nums[right] == 0:
                k -= 1
            
            while k < 0:
                if nums[left] == 0:
                    k += 1

                left += 1

            res = max(res, right - left + 1)

        return res

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.go
func longestOnes(nums []int, k int) int {
    l, res := 0, 0

    for r := 0; r < len(nums); r++ {
        if nums[r] == 0 {
            k--
        }

        for k < 0 {
            if nums[l] == 0 {
                k++
            }

            l++
        }

        if res < r - l + 1 {
            res = r - l + 1
        }
    }  

    return res
}